src/rpc.rs

```diff
@@ -434,7 +527,7 @@ impl<T: Runtime> Rpc<T> {
         let events_sub = self.subscribe_events().await?;
         let mut xt_sub = self.watch_extrinsic(extrinsic).await?;
 
-        while let status = xt_sub.next().await {
+        while let Some(status) = xt_sub.next().await {
```

Comment:
we might run into https://github.com/paritytech/jsonrpsee/issues/220 when the subscription is dropped, should be fixed soon I have pending PR opened.